tracer/build/_build/PrepareRelease/GenerateIntegrationDefinitions.cs

```diff
@@ -114,24 +114,27 @@ static void WriteCallTargetDefinitionFile(StreamWriter swriter, IEnumerable<Call
             swriter.WriteLine("// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.");
             swriter.WriteLine("// </copyright>");
             swriter.WriteLine();
+            swriter.WriteLine("using System.Collections.Generic;");
+            swriter.WriteLine();
             swriter.WriteLine("namespace Datadog.Trace.ClrProfiler");
             swriter.WriteLine("{");
             swriter.WriteLine("    internal static partial class InstrumentationDefinitions");
             swriter.WriteLine("    {");
 
             // Default Integrations
             var normalCallTargetIntegrations = callTargetIntegrations.Where(i => i.IntegrationType == IntegrationType.Default).Distinct().ToList();
-            swriter.WriteLine("        private static NativeCallTargetDefinition[] GetDefinitionsArray()");
+            swriter.WriteLine("        private static List<NativeCallTargetDefinition> GetDefinitionsList()");
             swriter.WriteLine("        {");
-            swriter.WriteLine("            return new NativeCallTargetDefinition[]");
-            swriter.WriteLine("            {");
+            swriter.WriteLine("            List<NativeCallTargetDefinition> definitionList = new List<NativeCallTargetDefinition>();");
```

Comment:
If you want, you can keep a similar syntax when switching to `List<T>`:

```csharp
private static List<NativeCallTargetDefinition> GetDefinitionsArray()
{
    return new List<NativeCallTargetDefinition>()
    {
        // AdoNet
        new(...),
        new(...),
        ...
    };
}
```

So I think all you need to change is:
```diff
--- private static NativeCallTargetDefinition[] GetDefinitionsArray()
--- {
---     return new NativeCallTargetDefinition[]
+++ private static List<NativeCallTargetDefinition> GetDefinitionsArray()
+++ {
+++     return new List<NativeCallTargetDefinition>()
```

This syntax is just compiler sugar that ultimately compiles to the same thing you did. The compiler will add calls to `List<T>.Add(...)` for each item.

Reply:
thanks! ✅